### DISCUSSION TITLE
label_values plus empty string

### DISCUSSION THREAD

**Discussion by sandersaares** [grafana/grafana#38354] [Dashboards, Panels & Visualizations]
I want to be able to specify that an empty string is a valid value in addition to what is returned by `label_values` for a Prometheus metric, because some series may lack this label entirely, which is not represented by the results of `label_values` but can be queried for by providing an empty string.

For example:

```
persons_entered_shop_total{car = "Audi"} 5
persons_entered_shop_total{car = "Volkswagen"} 4
persons_entered_shop_total{car = ""} 8
```

The last row there is equivalent to lacking the "car" label entirely in the Prometheus data model, where empty label value means the label does not exist:

```
persons_entered_shop_total 8
```

In this case `label_values` would return "Audi" and "Volkswagen", giving me a filter between those two. But I also want an option in my filter that matches empty string, so I can see the third series.

An "Include 'None' option" checkbox similar to the existing "Include 'All' option" would be nice.

Obviously, I could work around this by a custom "none" value for the series (instead of empty string) but sometimes that's not practical or the metric generating component is not under my control.

**Comment by marefr** [comment]
Related to #19408

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Reply by epca-job** [reply]
How's it going

**Comment by epca-job** [comment]
I had a similar problem, Can anyone offer a solution?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
